plugins/woocommerce-blocks/assets/js/base/utils/address.ts

```diff
@@ -105,6 +105,32 @@ export const emptyHiddenAddressFields = <
 	return newAddress;
 };
 
+/**
+ * Sets fields to an empty string in an address.
+ *
+ * @param {Object} address The address to empty fields from.
+ * @return {Object} The address with all fields values removed.
+ */
+export const emptyAddressFields = <
+	T extends CartResponseBillingAddress | CartResponseShippingAddress
+>(
+	address: T
+): T => {
+	const addressForm = prepareFormFields(
+		ADDRESS_FORM_KEYS,
+		{},
+		address.country
+	);
+	const newAddress = Object.assign( {}, address ) as T;
+
+	addressForm.forEach( ( { key = '' } ) => {
+		if ( isValidAddressKey( key, address ) ) {
+			newAddress[ key ] = '';
```

Comment:
Will this successfully reset all type of [form elements](https://www.w3schools.com/html/html_form_elements.asp) and element types?

Reply:
Yes, this function will accept all elements because it is based on the address object rather than the form itself. The address object remains independent of the front-end display, whether it is a dropdown, radio button, checkbox, or any other element. It consists of properties in string format.
Type definitions:
 https://github.com/woocommerce/woocommerce/blob/ecae6df49cc7e6241463843e4fa9decd9c31b80d/plugins/woocommerce-blocks/assets/js/types/type-defs/cart-response.ts#L78-L83

https://github.com/woocommerce/woocommerce/blob/ecae6df49cc7e6241463843e4fa9decd9c31b80d/plugins/woocommerce-blocks/assets/js/types/type-defs/cart-response.ts#L30-L38